content/hi/docs/setup/production-environment/container-runtimes.md

```diff
@@ -0,0 +1,407 @@
+आप अपनी स्थापना को किसी विशिष्ट रिलीज़ पर पिन कर सकते हैं।
+संस्करण 1.20.0 स्थापित करने के लिए, `VERSION=1.20:1.20.0` सेट करें।
+<br />
+
+कृपया यह करें
+```shell
+sudo curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable.repo https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/devel:kubic:libcontainers:stable.repo
+sudo curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable:cri-o:$VERSION.repo https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:$VERSION/$OS/devel:kubic:libcontainers:stable:cri-o:$VERSION.repo
+sudo yum install cri-o
+```
+
+{{% /tab %}}
+
+{{% tab name="openSUSE Tumbleweed" %}}
+
+```shell
+sudo zypper install cri-o
+```
+{{% /tab %}}
+{{% tab name="Fedora" %}}
+
+`$VERSION` को CRI-O संस्करण में सेट करें जो आपके Kubernetes संस्करण से मेल खाता हो।
+उदाहरण के लिए, यदि आप CRI-O 1.20, `VERSION=1.20` स्थापित करना चाहते हैं।
+
+आप इसके साथ उपलब्ध संस्करण पा सकते हैं: 
+```shell
+sudo dnf module list cri-o
+```
+CRI-O फेडोरा पर विशिष्ट रिलीज के लिए पिनिंग का समर्थन नहीं करता है। 
+
+फिर इसे करें:
+```shell
+sudo dnf module enable cri-o:$VERSION
+sudo dnf install cri-o
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+इसे शुरू करें CRI-O:
+
+```shell
+sudo systemctl daemon-reload
+sudo systemctl enable crio --now
+```
+
+[CRI-O इंस्टॉलेशन गाइड](https://github.com/cri-o/cri-o/blob/master/install.md)
+
+
+#### cgroup driver
+
+CRI-O डिफ़ॉल्ट रूप से systemd cgroup ड्राइवर का उपयोग करता है।
+CRI-O uses the systemd cgroup driver per default.`cgroupfs` cgroup ड्राइवर पर स्विच करने के लिए, या तो `/etc/crio/crio.conf` संपादित करें या `/etc/crio/crio.conf.d/02-cgroup-manager.conf` में ड्रॉप-इन कॉन्फ़िगरेशन रखें। उदाहरण के लिए:  
+
+```toml
+[crio.runtime]
+conmon_cgroup = "pod"
+cgroup_manager = "cgroupfs"
+```
+कृपया बदले हुए `conmon_cgroup` पर भी ध्यान दें, जिसे `cgroupfs` के साथ CRI-O का उपयोग करते समय `पॉड` मान पर सेट करना होगा।
+आमतौर पर क्यूबलेट के cgroup ड्राइवर कॉन्फ़िगरेशन (आमतौर पर kubeadm के माध्यम से किया जाता है) और CRI-O को सिंक में रखना आवश्यक है। 
+
+### Docker
+
+1.अपने प्रत्येक नोड पर, अपने लिनक्स वितरण के लिए डॉकर को निम्नानुसार स्थापित करें
+[डॉकर इंजन स्थापित करें](https://docs.docker.com/engine/install/#server). 
+आप इसमें डॉकर का नवीनतम मान्य संस्करण पा सकते हैं [निर्भरता](https://git.k8s.io/kubernetes/build/dependencies.yaml) फ़ाइल.
+
+2.कंटेनर के समूह के प्रबंधन के लिए विशेष रूप से सिस्टमड का उपयोग करने के लिए डॉकर डेमॉन को कॉन्फ़िगर करें।
+   ```shell
+   sudo mkdir /etc/docker
+   cat <<EOF | sudo tee /etc/docker/daemon.json
+   {
+     "exec-opts": ["native.cgroupdriver=systemd"],
+     "log-driver": "json-file",
+     "log-opts": {
+       "max-size": "100m"
+     },
+     "storage-driver": "overlay2"
+   }
+   EOF
+   ```
+
+   {{< note >}}
+   `overlay2` is the preferred storage driver for systems running Linux kernel version 4.0 or higher, 
+   or RHEL or CentOS using version 3.10.0-514 and above.
```

Comment:
This note is not localized yet.

Reply:
Thank you for reviewing the changes, will update them soon and push